pkg/protocol/http/http.go

```diff
@@ -72,54 +70,12 @@ func newDefaultTransport() *http.Transport {
 	}
 }
 
-// ClientOpt is the argument of NewProtocolClient.
-// ClientOpt supports some opt by key, such as "http.transport", "tls.config".
-// if not set, default opt will be used.
-type ClientOpt struct {
-	opt map[string]interface{}
-}
-
-func NewClientOpt() *ClientOpt {
-	return &ClientOpt{
-		opt: make(map[string]interface{}),
-	}
-}
-
-func (opt *ClientOpt) Set(key string, value interface{}) error {
-	switch key {
-	case HTTPTransport:
-		if _, ok := value.(*http.Transport); !ok {
-			return errortypes.ErrConvertFailed
-		}
-		break
-	case TLSConfig:
-		if _, ok := value.(*tls.Config); !ok {
-			return errortypes.ErrConvertFailed
-		}
-		break
-	default:
-		return fmt.Errorf("not support")
-	}
-
-	opt.opt[key] = value
-	return nil
-}
-
-func (opt *ClientOpt) Get(key string) interface{} {
-	v, ok := opt.opt[key]
-	if !ok {
-		return nil
-	}
-
-	return v
-}
-
 var _ protocol.Client = &Client{}
 
 // Client is an implementation of protocol.Client for http protocol.
 type Client struct {
 	client    *http.Client
-	transport http.RoundTripper
+	transport *http.Transport
```

Comment:
This seems to be a bad switch to me. Do you have a strong reason of doing this?

I can see a lot of benefits of using `http.RoundTripper`, e.g a mocking roundtripper like [go-vcr][1].

[1]: https://github.com/dnaeon/go-vcr

Reply:
Good advice, now switch to http.RoundTripper again.